### PR TITLE
ocamlPackages.json-data-encoding: 0.10 -> 0.11

### PR DESCRIPTION
**Pull request by StridBot**
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://opam.ocaml.org/packages/json-data-encoding/json-data-encoding.0.11.

meta.description for ocamlPackages.json-data-encoding is: "Type-safe encoding to and decoding from JSON"

meta.homepage for ocamlPackages.json-data-encoding is: "https://gitlab.com/nomadic-labs/json-data-encoding"

meta.changelog for ocamlPackages.json-data-encoding is: ""

###### Updates performed
- Version update

###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- Warning: a test defined in `passthru.tests` did not pass

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 0.11 in filename of file in /nix/store/g9h5zsr7y4fn4k0vd8b8ibcb6nymq8xl-ocaml4.13.1-json-data-encoding-0.11

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
1 total rebuild path(s)

1 package rebuild(s)

0 x86_64-linux rebuild(s)
0 x86_64-darwin rebuild(s)


First fifty rebuilds by attrpath
ocamlPackages.json-data-encoding
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Build yourself:
```
nix-build -A ocamlPackages.json-data-encoding https://github.com/StridBot/nixpkgs/archive/353b0506836fe81354367aeee50f173fbadb0157.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/g9h5zsr7y4fn4k0vd8b8ibcb6nymq8xl-ocaml4.13.1-json-data-encoding-0.11
ls -la /nix/store/g9h5zsr7y4fn4k0vd8b8ibcb6nymq8xl-ocaml4.13.1-json-data-encoding-0.11/bin
```

---

</details>
<br/>

### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings

cc @ulrikstrid for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

### CODE REVIEW

**Review by ulrikstrid — APPROVED** [review]
Looks good, nixpkgs-review passes

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 165628` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>4 packages failed to build and are new build failures:</summary>
  <ul>
<li>ligo: log was empty</li>
<li>ocamlPackages.ezresto: log was empty</li>
<li>ocamlPackages.ezresto-directory: log was empty</li>
<li>ocamlPackages.resto-json: <a href="https://termbin.com/y2f6">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/7a6810ab7a73e7f496e73a98f769098f/raw/index.html">pretty log </a></li>
  </ul>
</details>
<details>
<summary>
34 packages built:
</summary>
<ul>
<li>ocamlPackages.data-encoding</li>
<li>ocamlPackages.json-data-encoding</li>
<li>ocamlPackages.json-data-encoding-bson</li>
<li>ocamlPackages.tezos-base</li>
<li>ocamlPackages.tezos-base-test-helpers</li>
<li>ocamlPackages.tezos-clic</li>
<li>ocamlPackages.tezos-client-base</li>
<li>ocamlPackages.tezos-context</li>
<li>ocamlPackages.tezos-crypto</li>
<li>ocamlPackages.tezos-error-monad</li>
<li>ocamlPackages.tezos-event-logging</li>
<li>ocamlPackages.tezos-event-logging-test-helpers</li>
<li>ocamlPackages.tezos-micheline</li>
<li>ocamlPackages.tezos-mockup-proxy</li>
<li>ocamlPackages.tezos-mockup-registration</li>
<li>ocamlPackages.tezos-p2p</li>
<li>ocamlPackages.tezos-p2p-services</li>
<li>ocamlPackages.tezos-protocol-environment</li>
<li>ocamlPackages.tezos-protocol-environment-structs</li>
<li>ocamlPackages.tezos-proxy</li>
<li>ocamlPackages.tezos-requester</li>
<li>ocamlPackages.tezos-rpc</li>
<li>ocamlPackages.tezos-rpc-http</li>
<li>ocamlPackages.tezos-rpc-http-client</li>
<li>ocamlPackages.tezos-rpc-http-client-unix</li>
<li>ocamlPackages.tezos-sapling</li>
<li>ocamlPackages.tezos-shell-context</li>
<li>ocamlPackages.tezos-shell-services</li>
<li>ocamlPackages.tezos-shell-services-test-helpers</li>
<li>ocamlPackages.tezos-signer-backends</li>
<li>ocamlPackages.tezos-signer-services</li>
<li>ocamlPackages.tezos-stdlib-unix</li>
<li>ocamlPackages.tezos-version</li>
<li>ocamlPackages.tezos-workers</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ocamlPackages.json-data-encoding:
<details>

An avoidable string conversion got detected: `    rev = "${version}";`
Please do not convert variables to a string without modifying them but use them directly instead.

</details>
</details>

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.resto-json

**Review by vbgl — CHANGES_REQUESTED** [review]
This PR is SPAM

**Comment by ulrikstrid** [comment]
> This PR is SPAM

Why is this  spam @vbgl? If it doesn't break anything it makes sense to keep packages up to date.

**Comment by vbgl** [comment]
https://logs.nix.ci/?key=nixos/nixpkgs.165628&attempt_id=fbf2d92e-ed4b-4466-893a-02b477b3023e

**Comment by SuperSandro2000** [comment]
> > This PR is SPAM
> 
> Why is this spam @vbgl? If it doesn't break anything it makes sense to keep packages up to date.

There are build regressions. Not sure how to fix them, I am not that familiar with ocaml.

**Comment by ulrikstrid** [comment]
`ezresto-{,-directory}` was removed in v0.7 of `resto` and is unused in this repository

**Review by ulrikstrid — APPROVED** [review]
This should be good to go now

File: pkgs/development/ocaml-modules/resto/default.nix [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
+  version = "0.7";
   src = fetchFromGitLab {
     owner = "nomadic-labs";
     repo = "resto";
     rev = "v${version}";
-    sha256 = "13h3zga7h2jhgbyda1q53szbpxcz3vvy3c51mlqk3jh9jq2wrn87";
+    sha256 = "sha256-aX7w/rsoOmbni8BOXa0WnoQ47Y5zl91vWvMobuNFT3Y=";
   };
 
   useDune2 = true;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/resto/default.nix:13]:
```suggestion
```

**Review by vbgl — COMMENTED** [review]
While you are at it, you can do some cleaning.

**Comment by ulrikstrid** [comment]
@vbgl, that has now been fixed thanks for pointing it out!

**Comment by vbgl** [comment]
This breaks `resto-acl` for OCaml < 4.07.

Moreover, it seems that the two commits are in the wrong order.

**Pull request merged**